packages/playwright-core/src/server/recorder.ts

```diff
@@ -246,8 +246,10 @@ export class Recorder implements InstrumentationListener, IRecorder {
   }
 
   private _refreshOverlay() {
-    for (const page of this._context.pages())
-      page.mainFrame().evaluateExpression('window.__pw_refreshOverlay()').catch(() => {});
+    for (const page of this._context.pages()) {
+      for (const frame of page.frames())
```

Comment:
i'm assuming that `page.frames()` returns all frames recursively, not just the top level ones